src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -344,97 +339,13 @@ protected internal virtual DurableOrchestrationClient GetClient(OrchestrationCli
                 attribute,
                 attr =>
                 {
-                    AzureStorageOrchestrationServiceSettings settings = this.GetOrchestrationServiceSettings(attr);
-
-                    AzureStorageOrchestrationService innerClient;
-                    if (this.orchestrationServiceSettings != null &&
-                        this.orchestrationService != null &&
-                        string.Equals(this.orchestrationServiceSettings.TaskHubName, settings.TaskHubName, StringComparison.OrdinalIgnoreCase) &&
-                        string.Equals(this.orchestrationServiceSettings.StorageConnectionString, settings.StorageConnectionString, StringComparison.OrdinalIgnoreCase))
-                    {
-                        // It's important that clients use the same AzureStorageOrchestrationService instance
-                        // as the host when possible to ensure we any send operations can be picked up
-                        // immediately instead of waiting for the next queue polling interval.
-                        innerClient = this.orchestrationService;
-                    }
-                    else
-                    {
-                        innerClient = new AzureStorageOrchestrationService(settings);
-                    }
-
+                    var innerClient = this.orchestrationServiceFactory.GetOrchestrationClient(attribute);

```

Comment:
I prefer we use actual class names in variable decorations when the right-hand expression doesn't already contain it. It makes the code a bit more understandable.